api/src/controllers/comments.ts

```diff
@@ -0,0 +1,199 @@
+import { ErrorCode, isDirectusError } from '@directus/errors';
+import type { PrimaryKey } from '@directus/types';
+import express from 'express';
+import { respond } from '../middleware/respond.js';
+import useCollection from '../middleware/use-collection.js';
+import { validateBatch } from '../middleware/validate-batch.js';
+import { CommentsService } from '../services/comments.js';
+import { MetaService } from '../services/meta.js';
+import asyncHandler from '../utils/async-handler.js';
+import { sanitizeQuery } from '../utils/sanitize-query.js';
+
+const router = express.Router();
+
+router.use(useCollection('directus_comments'));
+
+router.post(
+	'/',
+	asyncHandler(async (req, res, next) => {
+		const service = new CommentsService({
+			accountability: req.accountability,
+			schema: req.schema,
+		});
+
+		const savedKeys: PrimaryKey[] = [];
+
+		if (Array.isArray(req.body)) {
+			const keys = await service.createMany(req.body);
+			savedKeys.push(...keys);
+		} else {
+			const key = await service.createOne(req.body);
+			savedKeys.push(key);
+		}
+
+		try {
+			if (Array.isArray(req.body)) {
+				const records = await service.readMany(savedKeys, req.sanitizedQuery);
+				res.locals['payload'] = { data: records };
+			} else {
+				const record = await service.readOne(savedKeys[0]!, req.sanitizedQuery);
+				res.locals['payload'] = { data: record };
+			}
+		} catch (error: any) {
+			if (isDirectusError(error, ErrorCode.Forbidden)) {
+				return next();
+			}
+
+			throw error;
+		}
+
+		return next();
+	}),
+	respond,
+);
+
+const readHandler = asyncHandler(async (req, res, next) => {
+	const service = new CommentsService({
+		accountability: req.accountability,
+		schema: req.schema,
+	});
+
+	const metaService = new MetaService({
+		accountability: req.accountability,
+		schema: req.schema,
+	});
+
+	let result;
+
+	if (req.body.keys) {
+		result = await service.readMany(req.body.keys, req.sanitizedQuery);
+	} else {
+		result = await service.readByQuery(req.sanitizedQuery);
+	}
+
+	const meta = await metaService.getMetaForQuery('directus_comments', req.sanitizedQuery);
+
+	res.locals['payload'] = { data: result, meta };
```

Comment:
Following controller paradigms we should return null if no result
```suggestion
	res.locals['payload'] = { data: result || null, meta };
```